lib/pbench/server/cache_manager.py

```diff
@@ -226,6 +255,76 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def cache_map(self, tar_path: Path):
+        """Builds Hierarchy structure of a Directory in a Dictionary
+        Format.
+
+        Args:
+            dir_path: root of a directory
+        """
+        cmap = {tar_path.name: {"details": FileInfo(tar_path)}}
+        dir_queue = deque([(tar_path, cmap)])
+        while dir_queue:
+            tar_path, parent_map = dir_queue.popleft()
+            tar_n = tar_path.name
+            curr = parent_map[tar_n]
+
+            dir_list = list(tar_path.glob("*"))
+            if dir_list:
+                curr["children"] = {}
+                curr = curr["children"]
+
+            for l_path in dir_list:
+                tar_info = FileInfo(l_path)
+                curr[l_path.name] = {"details": tar_info}
+                if l_path.is_dir():
+                    dir_queue.append((l_path, curr))
+        self.cachemap = cmap
+
+    @staticmethod
+    def traverse_cmap(path: Path, cachemap: dict):
+        name = path.name
+        files_list = path.parts
+        c_map = cachemap
+
+        for file_l in files_list:
+            if file_l not in ("/", "//"):
```

Comment:
You're never altering `c_map` if `file_l` *is* `/` or `//`. So, in `["/", "var", "x"]`, which is the `.parts` for `/var/x`, we'll just run through `files_list` without changing anything and return the root `cachemap` ... which I don't think is what we want!